[ONNX] Add a new operator for fill_() function.

fatcat-z opened the pull request:
Add the corresponding symbolic function and test.

fatcat-z committed:
[ONNX] Add support for hann_window operator. (#54587)

* [ONNX] Improve index_put symbolic to handle singular Bool updates (#53690)

Adds support for cases where the updates to the index_put node is a single Bool value, such as the case shown below

```
mask[indices] = True
```

Fixes #53507

* [ONNX] Support primitive type input/outputs and attributes (#53550)

Support primitive type attributes. Needed for Silero model.

* [ONNX] Fix if output shape mismatch error & Fix graph input directly used as output (#53219)

Fix if output shape mismatch error & Fix graph input directly used as output

* Add support for hann_window operator.

* [ONNX] Replace decomposeLinear pre process pass with a symbolic (#53077)

Replace decomposeLinear pre process pass with a symbolic

* Add a test case for dtype is None.

* Resolve flake8 issue.

* Remove one unused test case.

* Add support for hann_window operator.

* Add a test case for dtype is None.

* Remove one unused test case.

Co-authored-by: Shubham Bhokare <32080845+shubhambhokare1@users.noreply.github.com>
Co-authored-by: Negin Raoof <neginmr@utexas.edu>
Co-authored-by: Bowen Bao <bowbao@microsoft.com>
Co-authored-by: Ksenija Stanojevic <KsenijaS@users.noreply.github.com>

fatcat-z committed:
[ONNX] Add support for hann_window operator. (#54587)

* [ONNX] Improve index_put symbolic to handle singular Bool updates (#53690)

Adds support for cases where the updates to the index_put node is a single Bool value, such as the case shown below

```
mask[indices] = True
```

Fixes #53507

* [ONNX] Support primitive type input/outputs and attributes (#53550)

Support primitive type attributes. Needed for Silero model.

* [ONNX] Fix if output shape mismatch error & Fix graph input directly used as output (#53219)

Fix if output shape mismatch error & Fix graph input directly used as output

* Add support for hann_window operator.

* [ONNX] Replace decomposeLinear pre process pass with a symbolic (#53077)

Replace decomposeLinear pre process pass with a symbolic

* Add a test case for dtype is None.

* Resolve flake8 issue.

* Remove one unused test case.

* Add support for hann_window operator.

* Add a test case for dtype is None.

* Remove one unused test case.

Co-authored-by: Shubham Bhokare <32080845+shubhambhokare1@users.noreply.github.com>
Co-authored-by: Negin Raoof <neginmr@utexas.edu>
Co-authored-by: Bowen Bao <bowbao@microsoft.com>
Co-authored-by: Ksenija Stanojevic <KsenijaS@users.noreply.github.com>

BowenBao committed:
[ONNX] Fix assign input shape for tuple inputs & primitive type inputs (#54112)

shubhambhokare1 committed:
[ONNX] Update repeat_interleave symbolic (#54312)

Add implementation for cases when
- interleaving happens along dim which consist of dynamic axes

fatcat-z committed:
Add a new operator for fill_ function.

fatcat-z committed:
Add a new operator for fill_ function.

fatcat-z committed:
Add a new operator for fill_ function.

fatcat-z committed:
Add a new operator for fill_ function.

fatcat-z committed:
[ONNX] Enable tensordot symbolic function. (#55654)

Support tensordot in symbolic function of opset 12, and add tests accordingly.

neginraoof commented on test/onnx/test_pytorch_onnx_onnxruntime.py:
What if the dtype for x and filled_val does not match? Do we need to cover this case?

BowenBao commented:
Please check and resolve simple test file conflict

skyline75489 committed:
Fix onnx/constant_fold.cpp compilation on Windows (#55770)

VC++ does not recognize `or` as a valid operator. This breaks the build under `Debug` configuration.

fatcat-z commented on test/onnx/test_pytorch_onnx_onnxruntime.py:
According to the behavior of torch.Tensor.fill_() method, it will keep the dtype of the original tensor so I've done similar work in the symbolic function by passing the dtype of input tensor into full_like() function to make sure the dtype of final outout is same as the input.

fatcat-z committed:
Merge branch 'onnx_ms_1' into add_fill

fatcat-z commented:
> Please check and resolve simple test file conflict

Done.

KsenijaS committed:
[ONNX] Add support for prim::min (#55259)

Add support for prim::min operator and update full_like

neginraoof committed:
[ONNX] Export mv op (#55470)

Adding matrix-vector multiplication op

neginraoof committed:
[ONNX] Export mv op (#55470)

Adding matrix-vector multiplication op

fatcat-z committed:
Merge branch 'onnx_ms_1' into add_fill

neginraoof commented:
@BowenBao @fatcat-z 
Based on a discussion we had with Jay earlier, it looks like fill_ should be handled in the RemoveMutation jit pass as a special op and be replaced by full_like: https://github.com/pytorch/pytorch/blob/3646fa3621acc9ee3f3a7956bfc544375e74fdb4/torch/csrc/jit/passes/remove_mutation.cpp#L33
Can we check why this because fill_.Scalar is handled, but fill_.Tensor is missed? We might not need the symbolic then.

neginraoof commented:
@fatcat-z Can we create an issue for this on github?

fatcat-z commented:
> @BowenBao @fatcat-z
> Based on a discussion we had with Jay earlier, it looks like fill_ should be handled in the RemoveMutation jit pass as a special op and be replaced by full_like:
> 
> https://github.com/pytorch/pytorch/blob/3646fa3621acc9ee3f3a7956bfc544375e74fdb4/torch/csrc/jit/passes/remove_mutation.cpp#L33
> 
> 
> Can we check why this because fill_.Scalar is handled, but fill_.Tensor is missed? We might not need the symbolic then.

Is remove_muta

> @BowenBao @fatcat-z
> Based on a discussion we had with Jay earlier, it looks like fill_ should be handled in the RemoveMutation jit pass as a special op and be replaced by full_like:
> 
> https://github.com/pytorch/pytorch/blob/3646fa3621acc9ee3f3a7956bfc544375e74fdb4/torch/csrc/jit/passes/remove_mutation.cpp#L33
> 
> 
> Can we check why this because fill_.Scalar is handled, but fill_.Tensor is missed? We might not need the symbolic then.

I made some code changes like below, and then we no longer need a new fill() method in symbolic function. Looks like this new rule is missed with an unknown reason. Is it better we submit such a change to master branch directly?

**_/repos/pytorch/torch/csrc/jit/passes/remove_mutation.h_**
 bool isSpecialMappedOp(Node* n) {
    return n->matches("aten::zero_(Tensor(a!) self) -> Tensor(a!)") ||
        n->matches(
            "aten::fill_.Scalar(Tensor(a!) self, Scalar value) -> Tensor(a!)") ||
        n->matches(
            "aten::fill_.Tensor(Tensor(a!) self, Tensor value) -> Tensor(a!)") ||
        n->matches(
            "aten::normal_(Tensor(a!) self, float mean=0, float std=1, *, Generator? generator=None) -> Tensor(a!)");
  }

**_/home/jiz/repos/fatcat-z/pytorch/torch/csrc/jit/passes/remove_mutation.cpp_**

Node* MutationRemover::createSpecialMappedOp(Node* n) {
  WithInsertPoint guard(n);
  auto inputs = n->inputs();
  Node* new_node;
  if (n->matches(
          "aten::fill_.Scalar(Tensor(a!) self, Scalar value) -> Tensor(a!)") ||
        n->matches(
          "aten::fill_.Tensor(Tensor(a!) self, Tensor value) -> Tensor(a!)")) {
    new_node =
        graph_->insert(aten::full_like, {inputs.at(0), inputs.at(1)})->node();
  } else if (n->matches("aten::zero_(Tensor(a!) self) -> Tensor(a!)")) {
    new_node = graph_->insert(aten::zeros_like, {n->inputs().at(0)})->node();
  } else if (
       ...

BowenBao commented:
Maybe because `full_like` only accept scalar `fill_value`?

https://github.com/pytorch/pytorch/blob/3646fa3621acc9ee3f3a7956bfc544375e74fdb4/aten/src/ATen/native/native_functions.yaml#L1738

`fill_.Tensor` was explicitly left out in the original PR.

https://github.com/pytorch/pytorch/blob/3646fa3621acc9ee3f3a7956bfc544375e74fdb4/test/jit/test_remove_mutation.py#L145

@eellison do you recall what was the reason behind this?

zasdfgbnm committed:
Enable skipped test for c10::complex on CUDA >= 11.2 (#50227)

Summary:
That test was skipped due to a compiler bug. That bug should be fixed in 11.2, so we should enable it.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/50227

Reviewed By: malfet

Differential Revision: D27909195

Pulled By: anjali411

fbshipit-source-id: c802702079d0e521f53fc98cd0fc3ded0c12b455

bdhirsh committed:
Reland of "D27708346: generate xla codegen in-tree" (#56601)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56601

Updating it to ensure that RegistrationDeclarations.yaml is completely
unchanged

This reverts commit 90e532f3ef17a9611e9e7a9f1f6189d4168bf084.

Test Plan: Imported from OSS

Reviewed By: ailzhang

Differential Revision: D27915305

Pulled By: bdhirsh

fbshipit-source-id: 491a025c44221690dad849f9a2166934130c0fec

beningodfrey4 committed:
Fix errors when initializing Linear with 0 in_features (#56505)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/48152

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56505

Reviewed By: malfet

Differential Revision: D27919590

Pulled By: jbschlosser

fbshipit-source-id: 462ca280051f63c31ff588c38a9e436116c0f336

malfet committed:
Disable Windows GPU testing (#56655)

Summary:
Until https://github.com/pytorch/pytorch/issues/56654 is resolved

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56655

Reviewed By: ilia-cher

Differential Revision: D27929002

Pulled By: malfet

fbshipit-source-id: af741d67e4c938f632afad29e675533e1fcb445d

fatcat-z committed:
Merge branch 'add_fill' of https://github.com/fatcat-z/pytorch into add_fill

malfet committed:
Revert D27866138: [ONNX] Redesign inplace conversion (#55033)

Test Plan: revert-hammer

Differential Revision:
D27866138 (https://github.com/pytorch/pytorch/commit/24ff92f76d9998455258257334fca490b558cbd8)

Original commit changeset: ab5c9188740c

fbshipit-source-id: b99bf5b12e109089ebd5748c1dc152c6af1cebdb

xta0 committed:
[iOS GPU][Design] Support multiple tensors as outputs (#56072)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56072

Currently, we don't support outputting more than one tensors on GPU. For example, if you do

```
auto x = at::rand(1,4,2,2).metal();
auto y = at::chunk(x,2,1);  //y is a tuple
auto output1 = y[0].cpu();
auto output2 = y[1].cpu();
```
In the example above, when it hits `y[0].cpu()`, the command buffer will be committed to move `y[0]` from GPU to CPU. By the time it hits `y[1].cpu()`, since the command buffer has already become invalid, the temporary image that lives in `output2` has been recycled. Thus, a runtime exception will be thrown.

The way we address it is using the observer pattern
1. Before we flush the command buffer, we'll notify its the observers(a.k.a MPSImageWrapper objects) who hold the temporary images.
2. When observers receive the notification, they'll turn the current temporary images into a static images.
3. Now, when `.cpu()` happens, the output tensor can just read the data directly from the static image generated in the above step.

You may be wondering does that have a hidden cost where all the intermediate tensors have hold unused static images? The answers is no. All intermediate tensors will be released once their reference counts become zero. Since the MetalTensorImpl is subclassing from the TensorImpl, we're overriding the release_resource method  which gives us a chance to release the underlying storage (textures and buffers) and remove observers from the command buffer. Therefore, once the intermediate tensors go away, the temporary images will be recycled immediately.
ghstack-source-id: 127079751

Test Plan:
- We'll be using `at::chunk` to test this in the following diffs, as it returns a tuple that contains multiple tensors.
- Sandcastle CI
- CircleCI

Reviewed By: dreiss

Differential Revision: D27165886

fbshipit-source-id: 290b0d77b1dc74990b25cbd0abb775df1ab47ca0

Commit:
Update Kineto submodule and use new metadata api (#56432)

Summary:
Update Kineto submodule and use new metadata api

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56432

Test Plan: CI

Reviewed By: chaekit

Differential Revision: D27871570

Pulled By: ilia-cher

fbshipit-source-id: 3556787f07a9c9e138666a62ee4cd23af6d7473b

pbelevich committed:
Split test_c10d_spawn.py to test_c10d_spawn_gloo.py,test_c10d_spawn_nccl.py (#56599)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56599

Test Plan: NA

Reviewed By: SciPioneer

Differential Revision: D27913955

fbshipit-source-id: 7206e589fb7d08c55d08a58a3d57dc3d210a795e

pbelevich committed:
Split test_c10d.py to test_c10d_common.py, test_c10d_gloo.py, test_c10d_nccl.py (#56598)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56598

Test Plan: NA

Reviewed By: SciPioneer

Differential Revision: D27913170

fbshipit-source-id: 3439d18141131b02d55f2ca399a4c795cba2b04b

bertmaher committed:
Use at::cpu in bench_approx (#56563)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56563

Test Plan: Imported from OSS

Reviewed By: navahgar

Differential Revision: D27902737

Pulled By: bertmaher

fbshipit-source-id: 66962671afbb093d5ae0b9308a401536c06ce8f5

ajyu committed:
[static runtime] binding for aten::div_out (#56653)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56653

Test Plan:
```
./buck-out/opt/gen/caffe2/caffe2/fb/predictor/ptvsc2_predictor_bench --scripted_model=/data/users/ansha/tmp/adfinder/aug_1x/210616848_0.predictor.disagg.local.local.pt --pt_inputs=/data/users/ansha/tmp/adfinder/aug_1x/210616848_0.predictor.disagg.input_data.container.pt --iters=500 --warmup_iters=500 --num_threads=1 --pt_enable_static_runtime=1 --pt_cleanup_activations=true --pt_enable_out_variant=1 --pt_optimize_memory=1 --compare_results=1 --do_profile=1 --adsfinder_compatibility=1
```

```
Time per node type:
        1.48563 ms.    35.9861%. fb::sigrid_transforms_torch_bind (1 nodes)
        0.92385 ms.    22.3783%. aten::linear (6 nodes)
       0.681066 ms.    16.4974%. aten::argmin (1 nodes)
       0.239311 ms.    5.79679%. aten::matmul (1 nodes)
       0.140157 ms.    3.39501%. fb::clip_ranges_gather_sigrid_hash_v3 (77 nodes)
      0.0951568 ms.    2.30497%. fb::clip_ranges_gather (263 nodes)
      0.0835801 ms.    2.02455%. aten::sub (1 nodes)
       0.054081 ms.       1.31%. aten::repeat (1 nodes)
      0.0424465 ms.    1.02818%. aten::norm (1 nodes)
      0.0389049 ms.   0.942389%. fb::batch_box_cox (1 nodes)
      0.0346992 ms.   0.840514%. aten::__getitem__ (506 nodes)
      0.0341335 ms.    0.82681%. prim::TupleUnpack (254 nodes)
      0.0306839 ms.   0.743252%. aten::sigmoid (2 nodes)
      0.0280489 ms.   0.679426%. aten::mul (3 nodes)
      0.0265321 ms.   0.642684%. fb::offsets_to_ranges (253 nodes)
      0.0207622 ms.    0.50292%. aten::pow (1 nodes)
      0.0202067 ms.   0.489465%. fb::simple_embedding_bag_sum (3 nodes)
      0.0195497 ms.    0.47355%. fb::casted_batch_one_hot_lengths (1 nodes)
      0.0184351 ms.   0.446551%. fb::concat_add_mul_replacenan_clip (1 nodes)
       0.016382 ms.    0.39682%. aten::sum (3 nodes)
      0.0158651 ms.   0.384299%. prim::TupleConstruct (1 nodes)
      0.0150918 ms.   0.365567%. prim::DictConstruct (2 nodes)
     0.00858005 ms.   0.207833%. aten::div (1 nodes)
     0.00810684 ms.   0.196371%. fb::sigrid_hash_precompute (1 nodes)
     0.00796325 ms.   0.192893%. static_runtime::to_copy (8 nodes)
     0.00782038 ms.   0.189432%. prim::ListConstruct (4 nodes)
      0.0057504 ms.   0.139291%. aten::contiguous (1 nodes)
      0.0044688 ms.   0.108247%. aten::narrow (4 nodes)
     0.00284054 ms.   0.068806%. aten::logit (1 nodes)
     0.00265049 ms.  0.0642024%. aten::add (1 nodes)
     0.00216242 ms.    0.05238%. aten::full (1 nodes)
     0.00207732 ms.  0.0503187%. aten::relu (1 nodes)
     0.00198412 ms.   0.048061%. fb::gather_ranges (4 nodes)
     0.00176954 ms.  0.0428632%. aten::stack (1 nodes)
     0.00175913 ms.  0.0426112%. static_runtime::reshape_copy (2 nodes)
      0.0016996 ms.  0.0411692%. aten::clamp_min (1 nodes)
     0.00128528 ms.  0.0311331%. aten::size (3 nodes)
    0.000849156 ms.   0.020569%. aten::expand_as (1 nodes)
    0.000757672 ms.   0.018353%. fb::clip_ranges (2 nodes)
    0.000596224 ms.  0.0144423%. fb::lengths_to_offsets (3 nodes)
    0.000442632 ms.  0.0107218%. static_runtime::flatten_copy (1 nodes)
    0.000196158 ms. 0.00475151%. prim::device (1 nodes)
        4.12833 ms. in Total
StaticRuntime setup time: 0.000451 ms
Memory allocation time: 0.0089336 ms
Memory deallocation time: 0.0578358 ms
Outputs deallocation time: 0.0431742 ms
Total memory managed: 947328 bytes
Total number of reused tensors: 31
W0421 16:56:34.220682 1522800 PyTorchPredictorContainer.cpp:200] Failed to load metadata file
W0421 16:56:34.220772 1522800 PyTorchPredictorContainer.cpp:457] Couldn't find model param config file xl_model_weights/model_param_config
I0421 16:56:34.220791 1522800 PyTorchPredictorBenchLib.cpp:137] PyTorch predictor: number of prediction threads 1
I0421 16:56:34.366667 1522800 PyTorchPredictorBenchLib.cpp:230] PyTorch run finished. Milliseconds per iter: 145.863. Iters per second: 6.85573
I0421 16:56:34.514202 1522800 PtVsBlackBoxPredictorBenchLib.cpp:132] Finished comparing PT static runtime and jit interpreter results
```

Reviewed By: hlu1

Differential Revision: D27927731

fbshipit-source-id: 595883a31ba0cadf6449799d47bf2294a1d05b41

Chillee committed:
[FX] Make arg normalization a method on Node and not a pass (also augment tests to be exhaustive) (#55992)

Summary:
Commandeered from https://github.com/pytorch/pytorch/pull/54563

Primary changes from first PR:
1. Refactored primary `normalize_function` logic into `operator_schemas.py` so that non-FX users can use it.
2. Refactored tests a bit, and added a path to call `normalize_function` directly.
3. Moved check for `boolean_dispatch` so that `torch.lu` also gets properly handled.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/55992

Reviewed By: mruberry

Differential Revision: D27774396

Pulled By: Chillee

fbshipit-source-id: 7f65632e1d608e4abd55aec5ccbfdc3f67f52b8e

ysiraichi committed:
Port `scatter` and `scatter_add` to `OpInfo` (#56140)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/54302
Tracking Issue https://github.com/pytorch/pytorch/issues/54261

**Summary:**
- Port `scatter` and `scatter_add` tests to `OpInfo`
- `masked_scatter` was already ported to `OpInfo`

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56140

Reviewed By: malfet

Differential Revision: D27918038

Pulled By: heitorschueroff

fbshipit-source-id: 80b507fe8761cd15c967c85e0c289b568b877573

anjali411 committed:
Add OpInfo for torch.baddbmm (#56502)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56502

Reviewed By: heitorschueroff

Differential Revision: D27890939

Pulled By: anjali411

fbshipit-source-id: 072647a05cf93aedb76df0367af71b534be77258

Commit:
Fix bug in gaussian_nll_loss (#56469)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/53964. cc albanD almson

## Major changes:
- Overhauled the actual loss calculation so that the shapes are now correct (in functional.py)
- added the missing doc in nn.functional.rst

## Minor changes (in functional.py):
- I removed the previous check on whether input and target were the same shape. This is to allow for broadcasting, say when you have 10 predictions that all have the same target.
- I added some comments to explain each shape check in detail. Let me know if these should be shortened/cut.

Screenshots of updated docs attached.
Let me know what you think, thanks!

## Edit: Description of change of behaviour (affecting BC):
The backwards-compatibility is only affected for the `reduction='none'` mode. This was the source of the bug. For tensors with size (N, D), the old returned loss had size (N), as incorrect summation was happening. It will now have size (N, D) as expected.

### Example
Define input tensors, all with size (2, 3).
`input = torch.tensor([[0., 1., 3.], [2., 4., 0.]], requires_grad=True)`
`target = torch.tensor([[1., 4., 2.], [-1., 2., 3.]])`
`var = 2*torch.ones(size=(2, 3), requires_grad=True)`

Initialise loss with reduction mode 'none'. We expect the returned loss to have the same size as the input tensors, (2, 3).
`loss = torch.nn.GaussianNLLLoss(reduction='none')`

Old behaviour:
`print(loss(input, target, var)) `
`# Gives tensor([3.7897, 6.5397], grad_fn=<MulBackward0>. This has size (2).`

New behaviour:
`print(loss(input, target, var)) `
`# Gives tensor([[0.5966, 2.5966, 0.5966], [2.5966, 1.3466, 2.5966]], grad_fn=<MulBackward0>)`
`# This has the expected size, (2, 3).`

To recover the old behaviour, sum along all dimensions except for the 0th:
`print(loss(input, target, var).sum(dim=1))`
`# Gives tensor([3.7897, 6.5397], grad_fn=<SumBackward1>.`

![doc1](https://user-images.githubusercontent.com/26558092/115391089-f7f47b00-a1d6-11eb-8726-e4da9057aee0.png)
![doc2](https://user-images.githubusercontent.com/26558092/115391094-f925a800-a1d6-11eb-954b-afd187f42bc7.png)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56469

Reviewed By: jbschlosser, agolynski

Differential Revision: D27894170

Pulled By: albanD

fbshipit-source-id: 197890189c97c22109491c47f469336b5b03a23f

samestep committed:
Remove some wildcards from mypy configs (#56645)

Summary:
See https://github.com/pytorch/pytorch/pull/56523#issuecomment-823562134 for context. Basically the idea is that people (including myself) keep assuming that the single-asterisk `*` wildcard means "match in this directory and in its subdirectories", which is _not_ true. Removing the wildcards thus reduces confusion.

Ideally I would like to remove _all_ of these wildcards and then add a lint to disallow them in the future (and also greatly simplify the pattern-matching logic in `tools/mypy_wrapper.py`; see https://github.com/pytorch/pytorch/issues/55702 for context), but currently this one can't be removed:

```
tools/autograd/*.py,
```

That is because there is a file called `tools/autograd/templates/annotated_fn_args.py` (added in https://github.com/pytorch/pytorch/issues/41575) which is not a valid Python file and thus cannot be checked by `mypy`. ezyang would it be possible to rename that file to use a suffix other than `.py`?

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56645

Test Plan:
```
$ mypy
Success: no issues found in 1317 source files
$ mypy --config=mypy-strict.ini
Success: no issues found in 72 source files
```
The numbers of source files should be the same before and after this PR.

Reviewed By: ezyang

Differential Revision: D27925207

Pulled By: samestep

fbshipit-source-id: c17faf73665a75393d3109346a1138c2af023abb

malfet committed:
Fix quick-checks for operator-schemas (#56692)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56692

Reviewed By: heitorschueroff

Differential Revision: D27939830

Pulled By: malfet

fbshipit-source-id: 67a054de5c58832fcd7d0df0dd37faf1ea1406fd

samestep committed:
Always run all the grep-based quick-checks steps (#56700)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56700

Reviewed By: walterddr

Differential Revision: D27940638

Pulled By: samestep

fbshipit-source-id: 54311ef45ec051ee29d934d501e83b3542bbb439

ajyu committed:
[static runtime] binding for aten::sub_out (#56656)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56656

Test Plan:
```
./buck-out/opt/gen/caffe2/caffe2/fb/predictor/ptvsc2_predictor_bench --scripted_model=/data/users/ansha/tmp/adfinder/aug_1x/210616848_0.predictor.disagg.local.local.pt --pt_inputs=/data/users/ansha/tmp/adfinder/aug_1x/210616848_0.predictor.disagg.input_data.container.pt --iters=500 --warmup_iters=500 --num_threads=1 --pt_enable_static_runtime=1 --pt_cleanup_activations=true --pt_enable_out_variant=1 --pt_optimize_memory=1 --compare_results=1 --do_profile=1 --adsfinder_compatibility=1
```
```
Time per node type:
        1.85766 ms.    35.7817%. fb::sigrid_transforms_torch_bind (1 nodes)
         1.1238 ms.    21.6464%. aten::linear (6 nodes)
       0.858116 ms.    16.5288%. aten::argmin (1 nodes)
       0.334183 ms.    6.43694%. aten::matmul (1 nodes)
       0.173697 ms.     3.3457%. fb::clip_ranges_gather_sigrid_hash_v3 (77 nodes)
       0.118827 ms.    2.28881%. fb::clip_ranges_gather (263 nodes)
       0.101348 ms.    1.95215%. aten::sub (1 nodes)
      0.0748209 ms.    1.44118%. aten::repeat (1 nodes)
      0.0582576 ms.    1.12214%. aten::norm (1 nodes)
      0.0474353 ms.   0.913686%. fb::batch_box_cox (1 nodes)
      0.0457588 ms.   0.881393%. aten::__getitem__ (506 nodes)
      0.0435175 ms.   0.838222%. prim::TupleUnpack (254 nodes)
      0.0425416 ms.   0.819425%. aten::sigmoid (2 nodes)
      0.0383822 ms.   0.739308%. fb::offsets_to_ranges (253 nodes)
      0.0330187 ms.   0.635996%. aten::mul (3 nodes)
       0.027534 ms.   0.530352%. fb::simple_embedding_bag_sum (3 nodes)
      0.0274914 ms.   0.529532%. aten::pow (1 nodes)
      0.0236733 ms.   0.455989%. fb::casted_batch_one_hot_lengths (1 nodes)
       0.023348 ms.   0.449723%. fb::concat_add_mul_replacenan_clip (1 nodes)
      0.0193511 ms.   0.372735%. aten::sum (3 nodes)
      0.0188839 ms.   0.363737%. prim::DictConstruct (2 nodes)
      0.0183191 ms.   0.352858%. prim::TupleConstruct (1 nodes)
      0.0119029 ms.    0.22927%. aten::div (1 nodes)
      0.0103263 ms.   0.198902%. static_runtime::to_copy (8 nodes)
     0.00977658 ms.   0.188314%. prim::ListConstruct (4 nodes)
     0.00924042 ms.   0.177986%. fb::sigrid_hash_precompute (1 nodes)
     0.00692162 ms.   0.133322%. aten::contiguous (1 nodes)
     0.00567485 ms.   0.109307%. aten::narrow (4 nodes)
     0.00362285 ms.  0.0697823%. aten::logit (1 nodes)
     0.00329995 ms.  0.0635627%. aten::add (1 nodes)
     0.00285633 ms.  0.0550178%. aten::full (1 nodes)
     0.00268469 ms.  0.0517118%. fb::gather_ranges (4 nodes)
     0.00248577 ms.  0.0478803%. aten::stack (1 nodes)
     0.00241782 ms.  0.0465715%. aten::relu (1 nodes)
     0.00233674 ms.  0.0450096%. aten::clamp_min (1 nodes)
     0.00222238 ms.  0.0428068%. static_runtime::reshape_copy (2 nodes)
     0.00171177 ms.  0.0329716%. aten::size (3 nodes)
     0.00120008 ms.  0.0231155%. aten::expand_as (1 nodes)
     0.00112628 ms.  0.0216942%. fb::clip_ranges (2 nodes)
     0.00103193 ms.  0.0198768%. fb::lengths_to_offsets (3 nodes)
    0.000598624 ms.  0.0115305%. static_runtime::flatten_copy (1 nodes)
    0.000236196 ms. 0.00454954%. prim::device (1 nodes)
        5.19164 ms. in Total
StaticRuntime setup time: 0.000868 ms
Memory allocation time: 0.0109619 ms
Memory deallocation time: 0.071791 ms
Outputs deallocation time: 0.0560187 ms
Total memory managed: 1232320 bytes
Total number of reused tensors: 32
W0421 17:40:52.053653 1746499 PyTorchPredictorContainer.cpp:200] Failed to load metadata file
W0421 17:40:52.053757 1746499 PyTorchPredictorContainer.cpp:457] Couldn't find model param config file xl_model_weights/model_param_config
I0421 17:40:52.053779 1746499 PyTorchPredictorBenchLib.cpp:137] PyTorch predictor: number of prediction threads 1
I0421 17:40:52.185776 1746499 PyTorchPredictorBenchLib.cpp:230] PyTorch run finished. Milliseconds per iter: 131.985. Iters per second: 7.57661
I0421 17:40:52.337853 1746499 PtVsBlackBoxPredictorBenchLib.cpp:132] Finished comparing PT static runtime and jit interpreter results
```

Reviewed By: hlu1

Differential Revision: D27929253

fbshipit-source-id: 5a7984ba3ce2d6d4bce0a0ab6c5e09e8c037b44e

malfet committed:
Pass contiguous weight to NNPACK convolution (#56569)

Summary:
Added TestNN.test_conv2d_discontiguous_weight to prevent further regressions

Fixes https://github.com/pytorch/pytorch/issues/55781

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56569

Reviewed By: ngimel

Differential Revision: D27926509

Pulled By: malfet

fbshipit-source-id: fa5ce943c3e4db4aa4de1b1cba35bd399fb3c54d

subramen committed:
Add more model symbolic tracing tests from torchvision (#55744)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/55398

Generates tests that calls `symbolic_trace` on torchvision models and verifies the parity of outputs from eager model, `fx.GraphModule`, `jit.ScriptModule`.

Test errors: GoogleNet and Inception models throw a type mismatch when scripting the traced `fx.GraphModule`.
```
Return value was annotated as having type __torch__.torchvision.models.googlenet.GoogLeNetOutputs but is actually of type Tensor:
    dropout = self.dropout(flatten);  flatten = None
    fc = self.fc(dropout);  dropout = None
    return fc
    ~~~~~~~~~ <--- HERE
```

Relevant type-inconsistency https://github.com/pytorch/vision/blob/512ea299d4b2d2bbac3498a75a2d8c0190cfcb39/torchvision/models/googlenet.py#L200
```
torch.jit.unused
    def eager_outputs(self, x: Tensor, aux2: Tensor, aux1: Optional[Tensor]) -> GoogLeNetOutputs:
        if self.training and self.aux_logits:
            return _GoogLeNetOutputs(x, aux2, aux1)
        else:
            return x   # type: ignore[return-value]
```

Pull Request resolved: https://github.com/pytorch/pytorch/pull/55744

Reviewed By: albanD

Differential Revision: D27920595

Pulled By: suraj813

fbshipit-source-id: 01f6f2aef7badbde29b5162a7787b5af9398090d

wconstab committed:
Document current deploy cpython build #56490 (#56600)

Summary:
Call out the issues with cpython deps and suggest a workaround.

Fixes https://github.com/pytorch/pytorch/issues/56490

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56600

Reviewed By: albanD

Differential Revision: D27920647

Pulled By: wconstab

fbshipit-source-id: 61a53a176eaf42a6166d649d3cb0fdfa2489e9d2

garymm committed:
README: Minor improvements (#56193)

Summary:
* Visual studio versions: clarify and shorten.
* Remove obsolete note about a bug that has been fixed.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56193

Reviewed By: albanD

Differential Revision: D27939766

Pulled By: ezyang

fbshipit-source-id: e142ec04ba98d5468f28ddf2e8bba5d99d3cfc26

ejguan committed:
[DataLoader][doc] Randomness for base_seed generator and NumPy seed (#56528)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56528

Tried to search across internal and external usage of DataLoader. People haven't started to use `generator` for `DataLoader`.

Test Plan: Imported from OSS

Reviewed By: albanD

Differential Revision: D27908487

Pulled By: ejguan

fbshipit-source-id: 14c83ed40d4ba4dc988b121968a78c2732d8eb93

ejguan committed:
[DataLoader] Add Numpy seeding to worker of DataLoader (#56488)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56488

Considering amount of requests for this feature, introduce numpy seeding as default within each worker for DataLoader.

## BC-breaking Note:
- By introducing default numpy.random seeding strategy to workers of DataLoader, users don't need to manually set seed for workers by the `worker_init_fn`. And this PR won't influence users who are currently using `worker_init_fn` to set customized seed for workers.
- DataLoader will preserve reproducibility for users who are using numpy.random within Dataset.
- Multiprocessing (without `worker_init_fn` to define seed for numpy)
  - Start method as `spawn`: Each worker will now have seed for numpy random, rather than the seed generated from the imported time of Numpy module that make the DataLoader lose the reproducibility.
  - Start method as `fork`: Each worker not only have the same benefit as `spawn`,  but also have different seed for numpy as default, rather than inheriting the same seed.

Using the following Dataset and script as an example:
```py
class RandomDataset(Dataset):
    def __getitem__(self, ind):
        item = [ind, np.random.randint(1, 10000)]
        return item

    def __len__(self):
        return 20

if __name__ == '__main__'"
    ctx = mp.get_context('fork')
    ds = RandomDataset()
    g = torch.Generator()
    g.manual_seed(0)
    dl = DataLoader(ds, 2, shuffle=False, num_workers=4, multiprocessing_context=ctx, generator=g)

    epochs = 2
    for _ in range(epochs):
        for batch in d;:
            print(batch)
        print("====" * 10)
```

### 1.8.1:
Each worker generates same random result per iteration. And the seed will be reset to same for each epoch.
```py
tensor([[   0, 7449],
        [   1, 1519]])
tensor([[   2, 7449],
        [   3, 1519]])
tensor([[   4, 9645],
        [   5, 2387]])
tensor([[   6, 9645],
        [   7, 2387]])
tensor([[   8, 3118],
        [   9, 4552]])
=========================
tensor([[   0, 7449],
        [   1, 1519]])
tensor([[   2, 7449],
        [   3, 1519]])
tensor([[   4, 9645],
        [   5, 2387]])
tensor([[   6, 9645],
        [   7, 2387]])
tensor([[   8, 3118],
        [   9, 4552]])
=========================
```

### This PR:
Each worker has different seed at the beginning and re-seed for each epoch.
```py
tensor([[   0, 8715],
        [   1, 5555]])
tensor([[   2, 6379],
        [   3, 1432]])
tensor([[   4, 3271],
        [   5, 5132]])
tensor([[   6, 4287],
        [   7, 1104]])
tensor([[   8, 8682],
        [   9, 1699]])
=========================
tensor([[   0, 1374],
        [   1,  996]])
tensor([[   2,  143],
        [   3, 3507]])
tensor([[   4, 5887],
        [   5, 4730]])
tensor([[   6, 7274],
        [   7,  738]])
tensor([[   8, 6374],
        [   9, 1572]])
=========================
```

Test Plan: Imported from OSS

Reviewed By: albanD

Differential Revision: D27908486

Pulled By: ejguan

fbshipit-source-id: 5f313a30563bedeb88be214fa4beca0cefe9e4f4

seemethere committed:
Add RELEASE.md (#56520)

Summary:
The purpopse of this document is to outline our current release process
so that users coming into the project have a better idea on how the
actual release process works and how they can help contribute to it.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56520

Reviewed By: janeyx99

Differential Revision: D27890571

Pulled By: seemethere

fbshipit-source-id: 882a565ea8d9b9a46c9242be7cf79dede2bae63f

Commit:
[torch.Package] Folder has_file() method (#56584)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56584

Test Plan: Imported from OSS

Reviewed By: suo

Differential Revision: D27909314

Pulled By: Lilyjjo

fbshipit-source-id: facc89735ab67c87f0ec7653d8ccc359f98d4e0d

ezyang committed:
Port all non-float unary operators to structured (and rsqrt) (#56151)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56151

I missed rsqrt in the last PR.  The native_functions.yaml
was done with the following script:

```
import ruamel.yaml
from ruamel.yaml.tokens import CommentToken
from ruamel.yaml.error import CommentMark
from tools.codegen.model import *  # noqa: F403

with open("aten/src/ATen/native/native_functions.yaml", "r") as f:
    contents = f.read()

yaml = ruamel.yaml.YAML()
yaml.preserve_quotes = True
yaml.width = 1000
yaml.boolean_representation = ['False', 'True']
r = yaml.load(contents)

convert = '''\
rsqrt
bitwise_not
frac
i0
round
'''.split()

for e in r:
    f = NativeFunction.from_yaml(e, Location("", 0))
    if f.structured or f.structured_delegate is not None:
        continue
    n = f.func.name.name.base
    if n not in convert:
        continue
    # mutate e to make changes
    if f.func.kind() == SchemaKind.out:
        e.insert(1, 'structured', True)
        e.insert(2, 'structured_inherits', 'TensorIteratorBase')
    else:
        # TODO: The .out overload assumption is not sound in general
        e.insert(1, 'structured_delegate', f'{n}.out')

        if 'dispatch' in e:
            e['dispatch'].pop('CPU', None)
            e['dispatch'].pop('CUDA', None)
            e['dispatch'].pop('CPU, CUDA', None)
            e['dispatch'].pop('CompositeExplicitAutograd', None)
        else:
            print(n)

        *_, last_k = e.keys()
        needs_fixup = False

        if 'dispatch' in e and not e['dispatch']:
            if last_k == 'dispatch':
                needs_fixup = True
            del e['dispatch']

        # Manually fix up newlines at the end, because ruamel
        # made some bad life choices about where to associate trailing
        # whitespace for nested dicts; see
        # https://stackoverflow.com/questions/42172399/modifying-yaml-using-ruamel-yaml-adds-extra-new-lines
        if needs_fixup:
            *_, last_k = e.keys()
            # post_key, pre_key, post_value, pre_value
            e.ca.items[last_k] = [None, None, CommentToken('\n\n', CommentMark(0), None), None]

with open("aten/src/ATen/native/native_functions.yaml.new", "w") as f:
    yaml.dump(r, f)
```

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Reviewed By: bhosmer

Differential Revision: D27795078

Pulled By: ezyang

fbshipit-source-id: c8961b58753c12f985d786eae73f776c39d30e6e

bertmaher committed:
Disable TestComplexity.test_nn_module_test in fbcode (#56677)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56677

This has been failing with `RecursionError: maximum recursion depth
exceeded while calling a Python object` in fbcode for a while now.  Obviously
this isn't a fix, but the test works in OSS, so...
ghstack-source-id: 127146338

Test Plan:
```
buck test mode/dev //caffe2/test:jit -- --exact 'caffe2/test:jit - test_nn_module_tests (jit.test_complexity.TestComplexity)' --run-disabled
```

Reviewed By: Lilyjjo

Differential Revision: D27934963

fbshipit-source-id: 21d9858dab9ca1ebb5b67f286e788662dd24a988

Commit:
[c10d] Add requires_gloo decorator to test_logging_init (#56682)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56682

The process group is created on a gloo backend.

Context: https://github.com/pytorch/pytorch/pull/56598#discussion_r618206941
ghstack-source-id: 127150910

Test Plan: buck test caffe2/test/distributed:c10d -- test_logging_init

Reviewed By: pbelevich

Differential Revision: D27936805

fbshipit-source-id: 932efc638f94bdf78ddbae291e3720a20e43f2af

stas00 committed:
[doc] FX Graph Mode Quantization - fix preamble (#52192)

Summary:
The pre-amble here is misformatted at least and is hard to make sense of: https://pytorch.org/docs/master/quantization.html#prototype-fx-graph-mode-quantization

This PR is trying to make things easier to understand.

As I'm new to this please verify that my modifications remain in line with what may have been meant originally.

Thanks.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/52192

Reviewed By: ailzhang

Differential Revision: D27941730

Pulled By: vkuzo

fbshipit-source-id: 6c4bbf7c87d8fb87ab5d588b690a72045752e47a

driazati committed:
Re-order tests based on changed files (#56666)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56666

Addresses some of #56557 by checking for changed files when running tests. This will help deliver signal faster when a failing test is run. It should always be safe to at least try to re-order the tests, so there's no option to turn it off, and any error ends up bailing out of the sorting process. Time saved will change between tests, with more improvement for things that are further down the static list here:

https://github.com/pytorch/pytorch/blob/1e9c7ad4cb1869ea3769e1c563c78bce95da5945/test/run_test.py#L32

The results vary from not much improvement ([before: 11m](https://app.circleci.com/pipelines/github/pytorch/pytorch/307580/workflows/6ab3def6-8d63-4f41-9b8d-9c2c50f6266b/jobs/12712819/steps), [after: 10m](https://app.circleci.com/pipelines/github/pytorch/pytorch/307578/workflows/157407b4-f850-431c-b641-d2ac97916a04/jobs/12712802/steps)) to a lot ([before: 75m](https://app.circleci.com/pipelines/github/pytorch/pytorch/307580/workflows/6ab3def6-8d63-4f41-9b8d-9c2c50f6266b/jobs/12712884/steps), [after: 8m](https://app.circleci.com/pipelines/github/pytorch/pytorch/307578/workflows/157407b4-f850-431c-b641-d2ac97916a04/jobs/12712865/steps)), but overall there shouldn't be any regression in test timing. These results are also probably a little confounded since the test sharding will be different after re-ordering.

As a follow up we can use the target determination logic to figure out which tests to bring to front based on the actual code instead of just edits to test files

Test Plan: Imported from OSS

Reviewed By: samestep

Differential Revision: D27934076

Pulled By: driazati

fbshipit-source-id: 747d09ad732289d7693101803d46e9fa8e6d2f59

xta0 committed:
[iOS GPU] Fix Shader compilation errors for Metal 1.2 (iOS 12) (#56670)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56670

`int64_t` is only available for Metal 2.2 and above. `size_t` works fine in those situations. https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf
ghstack-source-id: 127169610

Test Plan:
- AIBench
```
buck run mode/mac aibench:run_bench_macos -- -b aibench/specifications/models/pytorch/metal/metal_unet_1001_detection.json --platform ios --framework pytorch --remote --devices D201 (https://github.com/pytorch/pytorch/commit/85b1c45a456591bafd51ca79b8a03cca59f90fde)AP-12.0.1
```

Reviewed By: linbinyu

Differential Revision: D27933297

fbshipit-source-id: 474b1eb191c68101367c9623c855645684434bd7

Commit:
Revert to ANVTM in jni_lite due to Oculus failure.

Test Plan: FanW123 verified on her Oculus device

Reviewed By: FanW123

Differential Revision: D27943428

fbshipit-source-id: ac1c1ca6b47937f8839ba23c9e3af0843ea086a3

mrshenli committed:
Re-enable test_device_maps_gpu (#56415)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56415

closes #53287

Test Plan: Imported from OSS

Reviewed By: lw

Differential Revision: D27865438

Pulled By: mrshenli

fbshipit-source-id: 3f7fcba8b799966388cc98ffc349cb62f281c367

peterbell10 committed:
Tell codegen that SparseCsrCUDA is cuda (#56602)

Summary:
Follow up to https://github.com/pytorch/pytorch/issues/50937, Fixes build failures in https://github.com/pytorch/pytorch/issues/56561

Currently SparseCsrCUDA is included in cpu build and also doesn't get code-generated device guards. This fixed both issues.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56602

Reviewed By: albanD

Differential Revision: D27921001

Pulled By: ezyang

fbshipit-source-id: 2b3b0b66d0a7c5ef96e0817d8852d511dd954ae4

Commit:
make beg_size output deterministic for EmbeddingBag (#56661)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56661

Under some conditions (requires_grad = false and mode=SUM) bag_size and max_indices will be created via at::empty and will not be modified, that is why corresponding outputs is not deterministic and causing tests to fail.

Test Plan: buck test mode/opt //caffe2/benchmarks/static_runtime:static_runtime_cpptest -- --exact 'caffe2/benchmarks/static_runtime:static_runtime_cpptest - StaticRuntime.EmbeddingBag' --run-disabled

Reviewed By: hlu1

Differential Revision: D27931445

fbshipit-source-id: fe9747094027e4e6f7c7b0771c1cd994f94fd554

janeyx99 committed:
Copy over test reports before running "report results" for linux test jobs (#56725)

Summary:
This way, if report results fail, the test reports are still saved as artifacts so we could use them to help us debug.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56725

Test Plan: CI linux test to pass + see that the test reports are copied in the Run tests step

Reviewed By: samestep

Differential Revision: D27948434

Pulled By: janeyx99

fbshipit-source-id: 597a2ba4fe1dca16c7b75a1399600b27f380f5cd

albanD committed:
Re-enable fast winograd conv on IOS (#56021)

Summary:
This is the proper fix for https://github.com/pytorch/pytorch/issues/38186
husthyc tested locally that it indeed fixes the issue there.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56021

Reviewed By: ailzhang

Differential Revision: D27940362

Pulled By: albanD

fbshipit-source-id: 020743315ce055633324ccd751c457e32ea3263d

soulitzer committed:
Fix grad_fn bindings when saved variable freed (#56499)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/54472

Adds HANDLE_TH_ERRORS to python bindings for grad_fn attrs and updates tests.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56499

Reviewed By: albanD

Differential Revision: D27920742

Pulled By: soulitzer

fbshipit-source-id: d4f7ac8c0aa2173d25517277c393f8c66de68951

janeyx99 committed:
Introducing JitPlugin (#56708)

Summary:
This PR is step 1 to covering JIT'd methods and functions. Step 2 (using it in CI) is here: https://github.com/pytorch/pytorch/issues/56310.

1. This PR introduces a package `coverage_plugins` that hosts JITPlugin.
2. We also bring in a `.coveragerc` file that is used in CI to omit the files we don't want to report on (e.g., temporary directories or test or utils.)

**Disclaimer: This PR does NOT use the plug-in. Nothing should change as a result.**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56708

Test Plan:
CI. Coverage should not go down.

If you're interested in testing this plug-in locally, you should:
`pip install -e tools/coverage_plugins_package` from the root directory.
Add the following lines to `.coveragerc` under `[run]`
```
plugins =
    coverage_plugins.jit_plugin
```
And then try:
`coverage run test/test_jit.py TestAsync.test_async_script_no_script_mod`

You should see `.coverage.jit` show up at the end. You can then run `coverage combine --append` and `coverage debug data` to see that some files in `torch/jit` are covered.

Reviewed By: samestep

Differential Revision: D27945570

Pulled By: janeyx99

fbshipit-source-id: 78732940fcb498d5ec37d4075c4e7e08e96a8d55

janeyx99 committed:
Remove extraneous variables in windows report stats step (#56596)

Summary:
Testing that the CIRCLE variables in the Windows test CI report stats step aren't needed.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56596

Test Plan: CI

Reviewed By: samestep

Differential Revision: D27948983

Pulled By: janeyx99

fbshipit-source-id: 71f2ca08246eea7580e31fb632612b205fb995fc

Commit:
[sparsity] Moving the sparsity python files to OSS (#56617)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56617

This migrates the sparsity to the open source

Test Plan: `buck test mode/opt //caffe2/test:ao`

Reviewed By: raghuramank100

Differential Revision: D27812207

fbshipit-source-id: cc87d9d2b486269901a4ad9b483615741a1cd712

malfet committed:
Add periodic_ prefix to all jobs run by cron (#56695)

Summary:
To make them more easily distinguishable in the HUD

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56695

Reviewed By: walterddr, samestep

Differential Revision: D27939938

Pulled By: malfet

fbshipit-source-id: e0abd1a6bc931a89f2aa5c6e2d8ebb471c461051

jerryzh168 committed:
[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule (#56550)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56550

Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Imported from OSS

Reviewed By: vkuzo, kazhang

Differential Revision: D27899317

fbshipit-source-id: ebf21334715e5ab764aaa27eed534cc0cdf9f2b5

Commit:
Create helper function for RPC profiling in _invoke_rpc and remote (#56643)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56643

Refactor enabling rpc profiling logic in `_invoke_rpc` and `remote()` into `_rpc_profiling()` helper function.

Reviewed By: rohan-varma

Differential Revision: D27922286

fbshipit-source-id: 27cfe662a401756f0ee8a3cd45978d933377f78f

jbschlosser committed:
Support factory kwargs in torch.nn modules (#54508)

Summary:
Continuation of https://github.com/pytorch/pytorch/pull/53144

Pull Request resolved: https://github.com/pytorch/pytorch/pull/54508

Reviewed By: albanD

Differential Revision: D27939544

Pulled By: jbschlosser

fbshipit-source-id: 4bf517e5f74f093e27ca38a85e732da65e44d805

Alvant committed:
Fix mp serialization for integer nn.Parameter on CUDA (#56529)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/56342

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56529

Reviewed By: albanD

Differential Revision: D27896094

Pulled By: ngimel

fbshipit-source-id: fe817781eb7139ea57c78acfd56e7c11b61eb4ed

seemethere committed:
torch: Add cpython as a dependency for torch_python_obj (#56740)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56740

Was running into a race condition where the torch_python_obj was
attempting to build before cpython was actually finished installing,
this should resolve that issue.

Only applicable on builds that use the `USE_DEPLOY=ON` option

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Test Plan: Imported from OSS

Reviewed By: malfet

Differential Revision: D27953782

Pulled By: seemethere

fbshipit-source-id: 76dd7c4218870eac97fc4c14e20b46128d264b30

Commit:
Print non-breaking space directly in lint.yml (#56726)

Summary:
After some fun investigating, samestep found that `\u1234` to produce a unicode character is only supported in bash > 4.2, but MacOS ship with bash/sh 3.2, so it was searching for the literal string `u1234`. This fixes the issue by printing out the char directly via its UTF-8 bytes and `printf`.
](https://our.intern.facebook.com/intern/diff/27952866/)
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56726

Pulled By: driazati

Reviewed By: SplitInfinity

Differential Revision: D27952866

fbshipit-source-id: 35871e959e250dfdbbdf8b121fc92212bc0614e8

soulitzer committed:
Clean up raise exception logic (#55656)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/55656

### For release notes
What:
 - All errors that are silenced by "raise_exception=False" are now GradcheckError (which inherits from RuntimeError).

Why:
 - Due to a refactor of gradcheck

Workaround:
 - If you catch for 'RuntimeError' with `except RuntimeError`, since GradcheckError inherits from RuntimeError, no changes are necessary. However if you explicitly check for the errors type via `type(error)`, you'll need to update your code to check for `GradcheckError` instead.

Factors out all the logic handling involving `fail_test`, `raise_exception` into 1) a wrapper around gradcheck that uses try/except 2) gradcheck_helper that always raises exception.
This allows us to avoid having to write the `if not x: return False` logic that is scattered throughout gradcheck currently.

Test Plan: Imported from OSS

Reviewed By: albanD

Differential Revision: D27920809

Pulled By: soulitzer

fbshipit-source-id: 253aef6d9a3b147ee37a6e37a4ce06437981929a

soulitzer committed:
Disable flaky tests (#56279)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56279

Test Plan: Imported from OSS

Reviewed By: albanD

Differential Revision: D27916606

Pulled By: soulitzer

fbshipit-source-id: 60c07024f6eb818f4aa6730a5f9ff90d7bc2b80f

soulitzer committed:
Enable fast gradcheck for real inputs and outputs (#55237)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/55237

In this PR, we reenable fast-gradcheck and resolve misc issues that arise:
Before landing this PR, land #55182 so that slow tests are still being run periodically.

Bolded indicates the issue is handled in this PR, otherwise it is handled in a previous PR.

**Non-determinism issues**:
- ops that do not have deterministic implementation (as documented https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms)
  - test_pad_cuda (replication_pad2d) (test_nn)
  - interpolate (test_nn)
  - cummin, cummax (scatter_add_cuda_kernel) (test_ops)
  - test_fn_gradgrad_prod_cpu_float64 (test_ops)

Randomness:
  - RRelu (new module tests) - we fix by using our own generator as to avoid messing with user RNG state (handled in #54480)

Numerical precision issues:
- jacobian mismatch: test_gelu (test_nn, float32, not able to replicate locally) - we fixed this by disabling for float32 (handled in previous  PR)
- cholesky_solve (test_linalg): #56235 handled in previous PR
- **cumprod** (test_ops) - #56275 disabled fast gradcheck

Not yet replicated:
 - test_relaxed_one_hot_categorical_2d (test_distributions)

Test Plan: Imported from OSS

Reviewed By: albanD

Differential Revision: D27920906

fbshipit-source-id: 894dd7bf20b74f1a91a5bc24fe56794b4ee24656

huiguoo committed:
[NNC] Add unroll and flatten APIs which not require return stmt pointer (#56420)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56420

Test Plan: Imported from OSS

Reviewed By: navahgar

Differential Revision: D27866118

Pulled By: huiguoo

fbshipit-source-id: f7e44fb20ef3a3c43b95d15f7b3b12e9e5cc89c9

malfet committed:
Move windows testers to previous image (#56626)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56626

Reviewed By: seemethere

Differential Revision: D27920812

Pulled By: malfet

fbshipit-source-id: faa739ca8500654df18cf963707b31c3345132cf

bertmaher committed:
CPU Convolution benchmark harness for some popular models (#56455)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56455

CPU convolution performance is pretty important for inference, so
tracking performance for CNNs often boils down to finding shapes that have
either regressed or need optimization.  This diff adds a benchmark harness that
lets you pretty easily add new sets of convolution parameters to benchmark.

I've started with an exhaustive list of layers from MobileNetV3, ResNet-18 and
ResNet-50, which are fairly popular torchvision models.  More to come if these
prove useful.

I've also added four backend configurations:

- native: uses at::conv2d, which applies its own backend selection heuristics
- mkldnn_none: uses mkldnn but applies no prepacking; uses the NCHW default
- mkldnn_weight: prepacks weights in an mkldnn-friendly format
- mkldnn_input: also prepacks the inputs in NCHW16c
ghstack-source-id: 127027784

Test Plan: Ran this on my Skylake Xeon

Reviewed By: ngimel

Differential Revision: D27876139

fbshipit-source-id: 950e1dfa09a33cc3acc7efd579f56df8453af1f2

BowenBao committed:
Revert "Revert [ONNX] Redesign inplace conversion" (#56675)

Summary:
Adjust how MutationRemover is used to avoid creating aliasDb multiple times for the same graph.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56675

Reviewed By: pbelevich

Differential Revision: D27945692

Pulled By: SplitInfinity

fbshipit-source-id: a6c548438e88ddee18ef03a6f0461ab9eaaaa829

Commit:
[torch.package] is_from_package check (#56729)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/56729

Test Plan: Imported from OSS

Reviewed By: suo

Differential Revision: D27949829

Pulled By: Lilyjjo

fbshipit-source-id: 1159d66d51b8f187c43847a5d449b13683c39eeb

kurtamohler committed:
Consolidate nondeterministic error tests (#55631)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/51498

Pull Request resolved: https://github.com/pytorch/pytorch/pull/55631

Reviewed By: malfet

Differential Revision: D27909953

Pulled By: mruberry

fbshipit-source-id: 9115b2433f9c276555be55bd51b270a7a2846829

Chillee committed:
moved more lowerings over (#55372)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/55372

Test Plan: Imported from OSS

Reviewed By: ZolotukhinM

Differential Revision: D27884601

Pulled By: Chillee

fbshipit-source-id: 91b00182abb5dcf60209425d2717fa0303cb4932

Commit:
static runtime support for fb::equally_split (#56565)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56565

fb::equally_split get fused with ListUnpack and all outputs from ListUnpack getting attached to fb::equally_split.
So fb::equal_split will have as many outputs as ListUnpack .

Test Plan:
buck test caffe2/torch/fb/sparsenn:fb_operators_test

buck test caffe2/torch/fb/sparsenn:test -- test_equally_split_op

Reviewed By: hlu1

Differential Revision: D27902824

fbshipit-source-id: 7855047c3bd46bbb74b7346ac384c70b6a3e1f46

aivanou committed:
Torchelastic: make process failure init error non-fatal (#56739)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56739

The diff makes several tiny changes:
* Add logs for each worker error file destination
* Make sure log_dir is propagated from the launcher
* Make ProcessFailure initialization error non-fatal.

Test Plan:
buck test mode/dev-nosan //caffe2/test/distributed/elastic/multiprocessing/errors:api_test

    https://fburl.com/tupperware/0nizb9z8

Reviewed By: borovsky-d, wilson100hong

Differential Revision: D27952596

fbshipit-source-id: 69582bf4be47758def4008f2abf82d123294cd1a

fatcat-z committed:
Merge branch 'add_fill' of https://github.com/fatcat-z/pytorch into add_fill
